### PR TITLE
#42 WIP Study static information

### PR DESCRIPTION
**Pull request by KijongHan**
I would like to get some feedback as to whether I am on the right track. 

-Added rest endpoint for retrieving all necessary information for "static study information" - contained within the StaticInfoDTO class
-Added faq field. I'm unsure how the static study information page is going to look so please give some suggestions

### CODE REVIEW

File: src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.project36.qualopt.service.dto;
+
+import org.project36.qualopt.domain.Study;
+
+public class StudyInfoDTO {
```
Comment (dylHall) [inline src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java]: Interesting, what else do you envision going into this StudyInfoDTO?

File: src/main/webapp/app/entities/study/study-detail.component.html [hunk 2]
```diff
@@ -19,6 +19,10 @@ <h2>{{study.name}}</h2>
         <dd>
             <span>{{study.emailBody}}</span>
         </dd>
+        <dt><span>FAQ</span></dt>
+        <dd>
+            <span>{{study.faq}}</span>
+        </dd>
```
Comment (dylHall) [inline src/main/webapp/app/entities/study/study-detail.component.html]: Will all the additional information be hosted in this special study page, or will it link to a separate page?

File: src/main/java/org/project36/qualopt/web/rest/StudyResource.java [hunk 3]
```diff
@@ -127,6 +128,21 @@ public StudyResource(StudyRepository studyRepository, StudyService studyService,
...
+     * @param id the id of the study, from which the study info will be retrieved
+     * @return the ResponseEntity with status 200 (OK) and with body the study, or with status 404 (Not Found)
+     */
+    @GetMapping("/studies/info/{id}")
+    @Timed
+    public ResponseEntity<StudyInfoDTO> getStudyInfo(@PathVariable Long id) {
+        log.debug("REST request to get study info: {}", id);
+        Study study = studyRepository.findOneWithEagerRelationships(id);
+        StudyInfoDTO studyInfo = new StudyInfoDTO(study);
+        return new ResponseEntity<>(studyInfo, HttpStatus.OK);
```
Comment (crat019) [inline src/main/java/org/project36/qualopt/web/rest/StudyResource.java]: Maybe as a precaution, add a case for when the study object was not found. Like returning a ResponseEntity object with a different status code.

File: src/main/webapp/app/entities/study/study-detail.component.html (hunk 2 above)
Comment (KijongHan) [reply src/main/webapp/app/entities/study/study-detail.component.html]: Definitely on a separate page as this special study page is only available to view to authenticated users. Study information is able to be viewed and edited by the user who created them.

File: src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java (hunk 1 above)
Comment (KijongHan) [reply src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java]: Issue #42 referenced also having an upload feature for uploading pdf/documents. I also envisioned other static site related things going into this DTO as well. Also, REST API for retrieving study entity requires authentication and didn't want to break that security. so instead added a security exception for retrieving a study info object as this needs to be viewed by Participants who won't be authenticated

File: src/main/java/org/project36/qualopt/web/rest/StudyResource.java (hunk 3 above)
Comment (KijongHan) [reply src/main/java/org/project36/qualopt/web/rest/StudyResource.java]: Yeap definitely. Thanks

File: src/main/webapp/app/entities/study/study-detail.component.html (hunk 2 above)
Comment (dylHall) [reply src/main/webapp/app/entities/study/study-detail.component.html]: Great that sounds good

File: src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java (hunk 1 above)
Comment (dylHall) [reply src/main/java/org/project36/qualopt/service/dto/StudyInfoDTO.java]: Yeah, that makes sense! Is there some Maven dependency (or just base java library) that can facilitate storage of PDFs? Not 100% sure if the current DB architecture would just be able to store them

**Comment by softeng-701** [comment]
@KijongHan Please stash your commits, thanks!

**Comment by softeng-701** [comment]
@KijongHan Please resolve the conflicts!

**Comment by KijongHan** [comment]
This PR implements the basic groundwork implementation outlined in issue #42. It involves both front-end and back-end changes.

## Back-end Changes

### Database/Entities
- A Document entity added to domain; stores Base64 encoded string of a file and other file attributes
- Study entity updated to include FAQ field and include a one-to-many relationship with Document entity
- Added a StudyInfoDTO entity (Not saved to database) that encapsulates all the necessary information from the Study entity that will be returned to the viewer of the "Study static information" (The idea is that developers can control what Study information is available and what should require authentication)

### REST API
- Add DocumentResource and add API endpoint for downloading a Document file
- Add API endpoint for retrieving StudyInfoDTO entity

### Security
- Add document download and study info request paths to security configuration pipeline so that they don't require authentication

## Front-end Changes

### Study Dialog Component
- Add a faq textarea field and a documents select field for the study creator to edit
- Can upload documents and remove documents

qualopt1

### Study Information Page
- Static page for study information, open publicly, showing basic information
- Can download documents from this page

qualopt2

**Comment by KijongHan** [comment]
@crat019 - Correct handling of "not found resources" has been addressed!
@dylHall - Document storage has been addressed!

@softeng-701 commits have been squashed

**Pull request merged**